syntax = "proto3";

package yandex.mdb.mlock.api.v1;

import "cloud/mdb/mlock/api/lock.proto";

option go_package = "a.yandex-team.ru/cloud/mdb/mlock/api;mlock";

service LockService {
  rpc ListLocks (ListLocksRequest) returns (ListLocksResponse) {}

  rpc GetLockStatus (GetLockStatusRequest) returns (LockStatus) {}

  rpc CreateLock (CreateLockRequest) returns (CreateLockResponse) {}

  rpc ReleaseLock (ReleaseLockRequest) returns (ReleaseLockResponse) {}
}

message ListLocksRequest {
  string holder = 1;

  int64 limit = 2;

  int64 offset = 3;
}

message ListLocksResponse {
  repeated Lock locks = 1;

  int64 next_offset = 2;
}

message GetLockStatusRequest {
  string id = 1;
}

message CreateLockRequest {
  string id = 1;

  string holder = 2;

  string reason = 3;

  repeated string objects = 4;
}

message CreateLockResponse {
}

message ReleaseLockRequest {
  string id = 1;
}

message ReleaseLockResponse {
}
